client/src/pages/Login.jsx

```diff
@@ -1,141 +1,79 @@
-  };
+const styles = {
+  card: {
+    width: 400,
+    mx: "auto",
+    p: 2,
+    mt: 4,
+    my: 4,
+    borderRadius: "sm",
+    display: "flex",
+    flexDirection: "column",
+    gap: 2,
+    boxShadow: "lg",
+    backgroundColor: "lightblue",
+  },
+};
 
-  const [login, { errors }] = useMutation(LOGIN);
-  const validateForm = () => {
-    let errors = {};
-    let isValid = true;
+const Login = () => {
+  const [login, { error }] = useMutation(LOGIN);
 
-    if (!formData.email.trim()) {
-      errors.email = "Please enter your email";
-      isValid = false;
-    }
+  const { loginSchema } = useSchemas();
 
-    if (!formData.password) {
-      errors.password = "Please enter your password";
-      isValid = false;
-    }
-    setFormErrors(errors);
-    return isValid;
-  };
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm({ resolver: yupResolver(loginSchema) });
 
-  const handleSubmit = async (event) => {
-    console.log("in handlesubmit");
-    event.preventDefault();
-    if (validateForm()) {
-      try {
-        const { data } = await login({
-          variables: { ...formData },
-        });
-        Auth.login(data);
-        alert("Successfully logged in!");
-      } catch (err) {
-        console.error(err);
-        alert(err);
-        alert("Login failed. Please try again.");
-      }
-      setFormData({
-        email: "",
-        password: "",
+  const onSubmit = async (userInput) => {
+    try {
+      const { data } = await login({
+        variables: { ...userInput },
       });
+      Auth.login(data);
+      alert("Successfully logged in!");
+    } catch (err) {
+      console.error(err);
+      alert("Login failed. Please try again.");
     }
   };
 
-  const handleClear = () => {
-    setFormData({
-      email: "",
-      password: "",
-    });
-    setFormErrors({
-      email: "",
-      password: "",
-    });
-  };
-
   return (
-    <Sheet
-      sx={{
-        width: 400,
-        mx: "auto",
-        p: 2,
-        mt: 4,
-        my: 4,
-        borderRadius: "sm",
-        display: "flex",
-        flexDirection: "column",
-        gap: 2,
-        boxShadow: "lg",
-        backgroundColor: "lightblue",
-      }}
-    >
-      <Typography level="h2" component="h1">
-        Login
-      </Typography>
-      <FormControl onSubmit={handleSubmit}>
-        <FormLabel htmlFor="email">Email</FormLabel>
-        <Input
-          type="email"
-          name="email"
-          id="email"
-          value={formData.email}
-          onChange={handleChange}
-        />
-        {formErrors.email && (
-          <Typography className="error">{formErrors.email}</Typography>
-        )}
-      </FormControl>
-      <FormControl>
-        <FormLabel htmlFor="password">Password</FormLabel>
+    <Card sx={styles.card}>
+      <Typography level="h2">Login</Typography>
+      <form onSubmit={handleSubmit(onSubmit)}>
```

Comment:
Is this really the pattern here? react-hook-forms provides `handleSubmit`, to which you must pass a function that actually handles submit? What, then, is it doing? Just validating with the schema stuff?

Reply:
This is indeed the pattern in the docs for react-hook-form. `handleSubmit()` is a wrapper that has additional functionality and options for handling the data. It adds the e.preventDefault() ....by default...🤷‍♂️ (sorry that feels like a bad pun). The second argument in handleSubmit is `onError` which give access to some validation errors. There are some other methods it has that I am not accessing so it may be the case that this is overkill here.